src/contact.rs

```diff
@@ -1094,4 +1096,104 @@ mod tests {
             vec![("Name one", "Address one"), ("Name two", "Address two")]
         )
     }
+
+    #[test]
+    fn test_get_contacts() {
+        let context = dummy_context();
+        let contacts = Contact::get_all(&context.ctx, 0, Some("some2")).unwrap();
+        assert_eq!(contacts.len(), 0);
+
+        let id = Contact::create(&context.ctx, "bob", "bob@mail.de").unwrap();
+        assert_ne!(id, 0);
+
+        let contacts = Contact::get_all(&context.ctx, 0, Some("bob")).unwrap();
+        assert_eq!(contacts.len(), 1);
+
+        let contacts = Contact::get_all(&context.ctx, 0, Some("alice")).unwrap();
+        assert_eq!(contacts.len(), 0);
+    }
+
+    #[test]
+    fn test_add_or_lookup() {
+        // add some contacts, this also tests add_address_book()
+        let t = dummy_context();
+        let book = "  Name one  \n one@eins.org \nName two\ntwo@deux.net\n\nthree@drei.sam\nName two\ntwo@deux.net\n";
```

Comment:
might be nicer to write with `concat!("foo\n", "bar\n")`.  up to you

Reply:
yip, this is nicer. i'll change that.